Source/WebInspectorUI/UserInterface/Views/TreeOutline.js

```diff
@@ -859,7 +859,7 @@ WI.TreeOutline = class TreeOutline extends WI.Object
         // In the no-word-wrap mode the outer <ol> may be wider than the tree container
         // (and partially hidden), in which case we use the edge of its container.
 
-        let scrollContainer = this.element.parentElement;
+        let scrollContainer = this.element.parentElement.parentElement;
```

Comment:
Are we sure this is safe to change here in such a specific class? You've only changed how a TreeOutline is contained by DOMTreeContentView, but many other places in Web Inspector use this class.

Reply:
Discussed with Patrick offline: I'll attempt to move the wrapper creation inside `WI.DOMTreeOutline` and expose a getter for the appropriate scroll container depending on context for `WI.TreeOutline` / `WI.DOMTreeOutline`.

Reply:
> Are we sure this is safe to change here in such a generic class? You've only changed how a TreeOutline is contained by DOMTreeContentView, but many other places in Web Inspector use this class.

@patrickangle Turns out I could use `width:fit-content` on the existing containing block, the  top-level `<ol>` created by `WI.DOMTreeOutline`, without introducing another wrapper or changing the scroll container.

I did an inspection of other uses of `WI.DOMTreeOutline` in Web Inspector and there is no visible impact elsewhere.

I did find an existing issue where their containers inhibit horizontal scroll and that prevents full expanding deeply nested node trees. At least one of them isn't a straightforward change and I don't want to block this patch on it. I filed https://bugs.webkit.org/show_bug.cgi?id=282348 as a separate issue.